File: app/src/main/java/com/fiap/inovagab/data/model/Ideia.kt
package com.fiap.inovagab.data.model

enum class StatusIdeia {
    ENVIADA,
    EM_ANALISE,
    APROVADA,
    REJEITADA,
    VIROU_PROJETO
}

enum class PrioridadeIdeia {
    BAIXA,
    MEDIA,
    ALTA
}

data class Ideia(
    val id: String = "",
    val titulo: String = "",
    val descricao: String = "",
    val area: String = "",
    val autorId: String = "",
    val autorNome: String = "",
    val status: StatusIdeia = StatusIdeia.ENVIADA,
    val prioridade: PrioridadeIdeia = PrioridadeIdeia.MEDIA,
    val estrategiaId: String = "",
    val versao: Int = 1,
    val criadoEm: Long = System.currentTimeMillis()
)

File: backend/src/InovaGAB.Domain/Ideias/Ideia.cs
namespace InovaGAB.Domain.Ideias;

public sealed class Ideia
{
    public string Id { get; set; } = string.Empty;

    public string Titulo { get; set; } = string.Empty;

    public string Descricao { get; set; } = string.Empty;

    public string Area { get; set; } = string.Empty;

    public string AutorId { get; set; } = string.Empty;

    public StatusIdeia Status { get; set; } = StatusIdeia.ENVIADA;

    public PrioridadeIdeia Prioridade { get; set; } = PrioridadeIdeia.MEDIA;

    public string EstrategiaId { get; set; } = string.Empty;

    public int EstrategiaVersao { get; set; }

    public int Versao { get; set; } = 1;

    public DateTime? ExcluidaEmUtc { get; set; }

    public DateTime CriadoEmUtc { get; set; }

    public DateTime AtualizadoEmUtc { get; set; }
}
